lib/spack/spack/directory_layout.py

```diff
@@ -149,6 +118,67 @@ def remove_install_directory(self, spec):
             path = os.path.dirname(path)
 
 
+class ExtensionsLayout(object):
+    """A directory layout is used to associate unique paths with specs for
+       package extensions.
+       Keeps track of which extensions are activated for what package.
+       Depending on the use case, this can mean globally activated extensions
+       directly in the installation folder - or extensions activated in
+       filesystem views.
+    """
+    def __init__(self, root, **kwargs):
+        self.root = root
+        self.link = kwargs.get("link", os.symlink)
+
+    def add_extension(self, spec, ext_spec):
+        """Add to the list of currently installed extensions."""
+        raise NotImplementedError()
+
+    def check_activated(self, spec, ext_spec):
+        """Ensure that ext_spec can be removed from spec.
+
+           If not, raise NoSuchExtensionError.
+        """
+        raise NotImplementedError()
+
+    def check_extension_conflict(self, spec, ext_spec):
+        """Ensure that ext_spec can be activated in spec.
+
+           If not, raise ExtensionAlreadyInstalledError or
+           ExtensionConflictError.
+        """
+        raise NotImplementedError()
+
+    def extension_map(self, spec):
+        """Get a dict of currently installed extension packages for a spec.
+
+           Dict maps { name : extension_spec }
+           Modifying dict does not affect internals of this layout.
+        """
+        raise NotImplementedError()
+
+    def extendee_target_directory(self, extendee):
+        """Specify to which full path extendee should link all files
+        from extensions."""
+        raise NotImplementedError
+
+    @property
+    def hidden_file_paths(self):
```

Comment:
I don't think either implementation of the interface implements this method. Is there some reason it is here?